oxidation/client/src/views/HomePage.vue

```diff
@@ -175,58 +249,153 @@ export interface DeviceStoredData {
 const { t, d } = useI18n();
 const deviceList: AvailableDevice[] = [
   {
-    organizationId: 'MegaShark',
-    humanHandle: 'Maxime Grandcolas',
+    organizationId: 'Planet Express Is The Best Comp!',
+    humanHandle: 'Dr. John A. Zoidberg',
     deviceLabel: 'device_label',
     keyFilePath: 'key_file_path',
     deviceId: 'device_id',
     slug: 'slug1',
     ty: {tag: 'Password'}
   },
   {
-    organizationId: 'Resana',
-    humanHandle: 'Maxime Grandcolas',
+    organizationId: 'PPTH',
+    humanHandle: 'Dr. Gregory House',
     deviceLabel: 'device_label',
     keyFilePath: 'key_file_path',
     deviceId: 'device_id',
     slug: 'slug2',
     ty: {tag: 'Password'}
   },
   {
-    organizationId: 'Oxymore',
-    humanHandle: 'Maxime Grandcolas',
+    organizationId: 'Black Mesa',
+    humanHandle: 'Dr. Gordon Freeman',
     deviceLabel: 'device_label',
     keyFilePath: 'key_file_path',
     deviceId: 'device_id',
     slug: 'slug3',
     ty: {tag: 'Password'}
   },
   {
-    organizationId: 'Eddy',
-    humanHandle: 'Maxime Grandcolas',
+    organizationId: 'OsCorp',
+    humanHandle: 'Dr. Otto G. Octavius',
     deviceLabel: 'device_label',
     keyFilePath: 'key_file_path',
     deviceId: 'device_id',
     slug: 'slug4',
     ty: {tag: 'Password'}
+  },
+  {
+    organizationId: 'Sanctum Sanctorum',
+    humanHandle: 'Dr. Stephen Strange',
+    deviceLabel: 'device_label',
+    keyFilePath: 'key_file_path',
+    deviceId: 'device_id',
+    slug: 'slug5',
+    ty: {tag: 'Password'}
+  },
+  {
+    organizationId: 'Holmes Consulting',
+    humanHandle: 'Dr John H. Watson',
+    deviceLabel: 'device_label',
+    keyFilePath: 'key_file_path',
+    deviceId: 'device_id',
+    slug: 'slug6',
+    ty: {tag: 'Password'}
+  },
+  {
+    organizationId: 'Riviera M.D.',
+    humanHandle: 'Dr. Nicholas "Nick" Riviera',
+    deviceLabel: 'device_label',
+    keyFilePath: 'key_file_path',
+    deviceId: 'device_id',
+    slug: 'slug7',
+    ty: {tag: 'Password'}
   }
 ];
 let selectedDevice: AvailableDevice;
 const password = ref('');
+const orgSearchString = ref('');
 const showOrganizationList = ref(true);
 const store = new Storage();
+const sortBy = ref('organization');
+const sortByAsc = ref(true);
+
+const msSelectOptions: MsSelectOption[] = [
+  { label: t('HomePage.organizationList.sortByOrganization'), key: 'organization' },
+  { label: t('HomePage.organizationList.sortByUserName'), key: 'user_name' },
+  { label: t('HomePage.organizationList.sortByLastLogin'), key: 'last_login' }
+];
+
+const msSelectSortByLabels = {
+  asc: t('HomePage.organizationList.sortOrderAsc'),
+  desc: t('HomePage.organizationList.sortOrderDesc')
+};
+
+const filteredDevices = computed(() => {
+  return deviceList.filter((item) => {
+    const lowerSearchString = orgSearchString.value.toLocaleLowerCase();
+    return (item.humanHandle?.toLocaleLowerCase().includes(lowerSearchString) ||
```

Comment:
is there any case where humanHandle is not set ?

Reply:
yes see:
export interface AvailableDevice {
    keyFilePath: Path;
    organizationId: OrganizationID;
    deviceId: DeviceID;
    humanHandle: HumanHandle | null;
    deviceLabel: DeviceLabel | null;
    slug: string;
    ty: DeviceFileType;
}